src/account/operations/output_consolidation.rs

```diff
@@ -22,36 +22,70 @@ const MIN_OUTPUT_SIZE_IN_ESSENCE: usize = 46;
 
 #[cfg(feature = "ledger_nano")]
 use crate::account::constants::DEFAULT_LEDGER_OUTPUT_CONSOLIDATION_THRESHOLD;
+use crate::Result;
 use crate::account::{
     constants::DEFAULT_OUTPUT_CONSOLIDATION_THRESHOLD, handle::AccountHandle,
-    operations::output_claiming::get_new_native_token_count, types::Transaction, TransactionOptions,
+    operations::{
+        helpers::time::can_output_be_unlocked_now,
+        output_claiming::get_new_native_token_count
+    },
+    types::{OutputData, Transaction},
+    AddressWithUnspentOutputs,
+    TransactionOptions,
 };
 
 impl AccountHandle {
+    fn should_consolidate_output(self: &AccountHandle, output_data: &OutputData, current_time: u32, account_addresses: &[AddressWithUnspentOutputs]) -> Result<bool> {
+        Ok(if let Output::Basic(basic_output) = &output_data.output {
+            let unlock_conditions = basic_output.unlock_conditions();
+
+            let is_time_locked = unlock_conditions.is_time_locked(current_time);
+            if is_time_locked {
+                // If the output is timelocked, then it cannot be consolidated.
+                return Ok(false);
+            }
+
+            let has_storage_deposit_return = unlock_conditions.storage_deposit_return().is_some();
+            let has_expiration = unlock_conditions.expiration().is_some();
+            let is_expired = unlock_conditions.is_expired(current_time);
+            if has_storage_deposit_return {
+                if has_expiration && !is_expired {
```

Comment:
Shouldn't this be
```suggestion
                if !has_expiration || (has_expiration && !is_expired) {
```
? Because without expiration we need to send the storage deposit back which we don't want